### ISSUE TITLE
[GetConnect] General request method

### ISSUE THREAD

**Issue by jasonlaw** [jonataslaw/getx#829]
Could you please add a general request method for GET, POST, PUT and DELETE?

eg: 
  request( "POST", url, body);

Thanks.

**Comment by jonataslaw** [comment]
Added on 3.21.0

**Comment by jasonlaw** [comment]
I don't think this is implemented correctly.
The method is never used and always trigger as put action.

~~~Dart

 @override
  Future<Response<T>> request<T>(
    String url,
    String method, {
    dynamic body,
    String contentType,
    Map<String, String> headers,
    Map<String, dynamic> query,
    Decoder<T> decoder,
  }) {
    _checkIfDisposed();
    return httpClient.put(
      url,
      body: body,
      headers: headers,
      contentType: contentType,
      query: query,
      decoder: decoder,
    );
  }

~~~

**Comment by eduardoflorence** [comment]
Hi @jasonlaw, fixed in 3.21.2
